webapp/src/cloud.ts

```diff
@@ -382,28 +393,27 @@ async function syncAsyncInternal(hdrs?: Header[]): Promise<Header[]> {
                         // No local changes, download latest.
                         const newHdr = await fromCloud(local, remoteFile);
                         pxt.tickEvent(`identity.sync.noConflict.localProjectUpdatedFromCloud`)
+                    } else if (local.cloudVersion === remoteFile.version) {
+                        // Local has unpushed changes, push them now
```

Comment:
If local and remote versions are the same, wouldn't that mean there are no local changes to push?

Reply:
This is really confusing, but `local.cloudVersion` is the last e-tag we received from the cloud. `local.cloudCurrent` says that there have been changes to the project since it was last saved to the cloud.

Reply:
Oh I see. We get here if cloudCurrent is false.

---
In reply to: [592894008](https://github.com/microsoft/pxt/pull/7947#discussion_r592894008) [](ancestors = 592894008)